packages/fxa-settings/src/pages/Signin/SigninTotpCode/container.tsx

```diff
@@ -0,0 +1,156 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { RouteComponentProps, useNavigate, useLocation } from '@reach/router';
+import { useValidatedQueryParams } from '../../../lib/hooks/useValidate';
+import { SigninQueryParams } from '../../../models/pages/signin';
+import { SigninTotpCode } from './index';
+import { gql, useMutation } from '@apollo/client';
+import GleanMetrics from '../../../lib/glean';
+import { logViewEvent } from '../../../lib/metrics';
+import { hardNavigate } from 'fxa-react/lib/utils';
+import {
+  AuthUiError,
+  AuthUiErrorNos,
+  AuthUiErrors,
+} from '../../../lib/auth-errors/auth-errors';
+import { currentAccount, sessionToken } from '../../../lib/cache';
+import { MozServices } from '../../../lib/types';
+import VerificationMethods from '../../../constants/verification-methods';
+import VerificationReasons from '../../../constants/verification-reasons';
+import { GraphQLError } from 'graphql';
+
+export const viewName = 'signin-totp-code';
+
+export type SigninTotpCodeContainerProps = {
+  serviceName: MozServices;
+};
+
+export const SigninTotpCodeContainer = ({
+  serviceName,
+}: SigninTotpCodeContainerProps & RouteComponentProps) => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const storedLocalAccount = currentAccount();
+
+  // TBD: A bit of a work around. In the previous signup step
+  // a new session token should have been generated. Setting
+  // it here to ensure that the authorization header is set
+  // when it comes time to verify the totp code.
+  if (storedLocalAccount?.sessionToken) {
+    sessionToken(storedLocalAccount?.sessionToken);
+  }
+
+  const { queryParamModel } = useValidatedQueryParams(SigninQueryParams);
+  const [verifyTotpCode] = useMutation(gql`
+    mutation VerifyTotp($input: VerifyTotpInput!) {
+      verifyTotp(input: $input) {
+        success
+      }
+    }
+  `);
+
+  const handleNavigation = () => {
+    // Note: `redirectTo` looks similar to `redirectPathname`, but they
+    // work slightly differently: `redirectTo` redirects automatically,
+    // while `redirectPathname` redirects after a form submission.
+
+    if (queryParamModel.redirectToUrl) {
+      hardNavigate(queryParamModel.redirectToUrl);
+    } else if (queryParamModel.redirectPathname) {
+      navigate(queryParamModel.redirectPathname + location.search);
+    } else if (
+      storedLocalAccount?.verificationReason ===
+      VerificationReasons.CHANGE_PASSWORD
+    ) {
+      navigate(`/post_verify/password/force_password_change${location.search}`);
```

Comment:
This needs a `hardNavigateToContentServer`. You could preemptively useConfig and check if postVerify routes are on or not as well, but we can also do that when we get to that epic.